backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/ShipmentSchedule.java

```diff
@@ -50,86 +50,37 @@
 @Builder
 public class ShipmentSchedule
 {
-	@NonNull
-	private final ShipmentScheduleId id;
-
-	@NonNull
-	private final OrgId orgId;
-
-	@NonNull
-	private final BPartnerId shipBPartnerId;
-
-	@NonNull
-	private final BPartnerLocationId shipLocationId;
-
-	@Nullable
-	private final BPartnerContactId shipContactId;
-
-	@Nullable
-	private final BPartnerId billBPartnerId;
-
-	@Nullable
-	private final BPartnerLocationId billLocationId;
-
-	@Nullable
-	private final BPartnerContactId billContactId;
-
-	@Nullable
-	private final OrderAndLineId orderAndLineId;
-
-	@Nullable
-	private final LocalDateTime dateOrdered;
-
-	private final LocalDate deliveryDateEffective;
-
+	@NonNull private final ShipmentScheduleId id;
+	@NonNull private final OrgId orgId;
+	@NonNull private final BPartnerId shipBPartnerId;
+	@NonNull private final BPartnerLocationId shipLocationId;
+	@Nullable private final BPartnerContactId shipContactId;
+	@Nullable private final BPartnerId billBPartnerId;
+	@Nullable private final BPartnerLocationId billLocationId;
+	@Nullable private final BPartnerContactId billContactId;
+	@Nullable private final OrderAndLineId orderAndLineId;
+	@Nullable private final LocalDateTime dateOrdered;
+	@Nullable private final LocalDate deliveryDateEffective;
 	private int numberOfItemsForSameShipment;
-
-	@NonNull
-	private final ProductId productId;
-
-	@NonNull
-	private final Quantity quantityToDeliver;
-
-	@NonNull
-	private final Quantity orderedQuantity;
-
-	@NonNull
-	private final Quantity deliveredQuantity;
-
-	@Nullable
-	private final AttributeSetInstanceId attributeSetInstanceId;
-
-	@NonNull
-	private APIExportStatus exportStatus;
-
-	@Nullable
-	private final ShipperId shipperId;
-
+	@NonNull private final ProductId productId;
+	@NonNull private final Quantity quantityToDeliver;
+	@NonNull private final Quantity orderedQuantity;
+	@NonNull private final Quantity deliveredQuantity;
+	@Nullable private final AttributeSetInstanceId attributeSetInstanceId;
+	@NonNull private APIExportStatus exportStatus;
+	@Nullable private final ShipperId shipperId;
 	private boolean isProcessed;
-
+	private boolean isClosed;
+	private boolean isActive;
 	@NonNull private CarrierAdviseStatus carrierAdvisingStatus;
-
 	@Nullable private String carrierAdviseErrorMessage;
-
 	@Nullable private CarrierProductId carrierProductId;
-
 	@Nullable private CarrierGoodsTypeId carrierGoodsTypeId;
-
 	@NonNull @Singular private Set<CarrierServiceId> carrierServices;
```

Comment:
IMHO 
* we need to distinguish between no carrier services and carrier services not loaded 
* i would use null to mark them as not loaded
* in that case we shall avoid Singular because that one is always produces non null fields
* i think we not generate a getter for them, but instead have getCarrierServicesIfLoaded() method